src/PricingModules/UniswapV3/UniswapV3WithFeesPricingModule.sol

```diff
@@ -0,0 +1,442 @@
+
+    /**
+     * @notice Checks for a token address and the corresponding Id if it is allow-listed.
+     * @param asset The contract address of the asset.
+     * @param assetId The Id of the asset.
+     * @return A boolean, indicating if the asset is whitelisted.
+     */
+    function isAllowListed(address asset, uint256 assetId) public view override returns (bool) {
+        if (!inPricingModule[asset]) return false;
+
+        try INonfungiblePositionManager(asset).positions(assetId) returns (
+            uint96,
+            address,
+            address token0,
+            address token1,
+            uint24,
+            int24,
+            int24,
+            uint128,
+            uint256,
+            uint256,
+            uint128,
+            uint128
+        ) {
+            return exposure[token0].maxExposure != 0 && exposure[token1].maxExposure != 0;
+        } catch {
+            return false;
+        }
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                          PRICING LOGIC
+    ///////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Returns the value of a Uniswap 3 Liquidity Range, denominated in USD.
+     * @param getValueInput A Struct with the input variables (avoid stack too deep).
+     * - asset: The contract address of the asset.
+     * - assetId: The Id of the range.
+     * - assetAmount: The amount of assets.
+     * - baseCurrency: The BaseCurrency in which the value is ideally denominated.
+     * @return valueInUsd The value of the asset denominated in USD, with 18 Decimals precision.
+     * @return valueInBaseCurrency The value of the asset denominated in a BaseCurrency different from USD, with 18 Decimals precision.
+     * @return collateralFactor The collateral factor of the asset for a given baseCurrency, with 2 decimals precision.
+     * @return liquidationFactor The liquidation factor of the asset for a given baseCurrency, with 2 decimals precision.
+     * @dev The UniswapV3PricingModule will always return the value denominated in USD.
+     */
+    function getValue(IPricingModule.GetValueInput memory getValueInput)
+        public
+        view
+        override
+        returns (uint256 valueInUsd, uint256, uint256 collateralFactor, uint256 liquidationFactor)
+    {
+        // Use variables as much as possible in local context, to avoid stack too deep errors.
+        address asset = getValueInput.asset;
+        uint256 id = getValueInput.assetId;
+        uint256 baseCurrency = getValueInput.baseCurrency;
+        address token0;
+        address token1;
+        uint256 usdPriceToken0;
+        uint256 usdPriceToken1;
+        uint256 principal0;
+        uint256 principal1;
+        {
+            int24 tickLower;
+            int24 tickUpper;
+            uint128 liquidity;
+            (,, token0, token1,, tickLower, tickUpper, liquidity,,,,) = INonfungiblePositionManager(asset).positions(id);
```

Comment:
what's the diff with saving those params?

Reply:
I don't get the question